src/schema/schema.graphql

```diff
@@ -509,6 +509,197 @@ type FreeBusy {
 	free: [FreeBusyInstance]
 }
 
+##### Filters ######
+
+enum Importance {
+	high
+	normal
+	low
+}
+
+enum FilterMatchCondition {
+	allof
+	anyof
+}
+
+type AddressCondition {
+	header: String!
+	part: String!
+	stringComparison: String!
+	caseSensitive: Boolean
+	value: String!
+	valueComparison: String
+	countComparison: String
+	index: Int
+	negative: Boolean
+}
+
+type BasicCondition {
+	index: Int
+	negative: Boolean
+}
+
+type BodyCondition {
+	caseSensitive: Boolean
+	value: String
+	index: Int
+	negative: Boolean
+}
+
+type ConversationCondition {
+	where: String
+	index: Int
+	negative: Boolean
+}
+
+type DateCondition {
+	dateComparison: String
+	date: Float # d
+	index: Int
+	negative: Boolean
+}
+
+type FlagCondition {
+	flagName: String!
+	index: Int
+	negative: Boolean
+}
+
+type HeaderCheckCondition {
+	header: String!
+	index: Int
+	negative: Boolean
+}
+
+type HeaderCondition {
+	header: String
+	stringComparison: String
+	valueComparison: String
+	countComparison: String
+	value: String
+	caseSensitive: Boolean
+	index: Int
+	negative: Boolean
+}
+
+type ImportanceCondition {
+	importance: Importance! # imp
+	index: Int
+	negative: Boolean
+}
+
+type InviteCondition {
+	methods: [String]
+	index: Int
+	negative: Boolean
+}
+
+type MimeHeaderCondition {
+	header: String
+	stringComparison: String
+	value: String
+	caseSensitive: Boolean
+	index: Int
+	negative: Boolean
+}
+
+type SizeCondition {
+	numberComparison: String
+	size: String # s
+	index: Int
+	negative: Boolean
+}
+
+type FilterCondition {
+	allOrAny: FilterMatchCondition! # condition
+
+	addressBook: [HeaderCheckCondition] # addressBookTest
```

Comment:
Seems odd to me that it is called a "test" in the API, but we call it a "condition" in the GraphQL schema

Reply:
in api we are having this structure

```
filterConditions 
    |-- headerTest
    |-- addressbookTest
```

which I have changed to

```
conditions
    |-- header
    |-- addressbook
```

This is done for better semantics, nothing else